### PR TITLE
Roles manager

### PR DESCRIPTION
**Pull request by mattbrailsford**
Setup an umbraco Roles Manager to work the same way the user manager does

### CODE REVIEW

File: src/UmbracoIdentity.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.23107.0
+# Visual Studio 2013
```
Comment (Shazwazza) [inline src/UmbracoIdentity.sln]: I'll prob revert the changes made in this file, not sure about that ProjectReferences part either.

**Comment by Shazwazza** [comment]
awesome! This looks super great! I'll have to gets to testing

File: src/UmbracoIdentity.sln (hunk 1 above)
Comment (mattbrailsford) [inline src/UmbracoIdentity.sln]: Yea, that's fine. I think I did the project ref by accident anyways :)

**Comment by rasmuseeg** [comment]
Would it be possible to add get all roles for a user and get all roles methods?

**Comment by Aventrue** [comment]
Hey guys, you really do an amazing great job with this project, it safed me a lot of time! Thank you very much! 
I have a little question. Is it possible to use then [Authorize(Roles = "RoleA,RoleB,RoleC")] ?

**Comment by Shazwazza** [comment]
@Aventrue yes, using the `AuthorizeAttribute`  with any of those parameters will always work. The AuthorizeAttribute is using 'AuthZ' (Authorization), all this is doing is checking what information is supplied by the current Principal of the request, the Principal is set with 'AuthN' (Authentication) at the beginning of the request. Therefore, if the Principal is created and assigned roles, then ` [Authorize(Roles = "RoleA,RoleB,RoleC")]` will certainly work.

BTW, I'm back from holidays now so will look at reviewing this code this week!

**Pull request merged**